api/test/e2e/route_with_vars_test.go

```diff
@@ -0,0 +1,307 @@
+					"vars": [
+						["http_k","==","header"]
+					],
+					"upstream": {
+						"type": "roundrobin",
+						"nodes": [{
+							"host": "172.16.238.20",
+							"port": 1980,
+							"weight": 1
+						}]
+					}
+				}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+
+		{
+			caseDesc:     "hit the route with right header",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"k": "header"},
+			Path:         `/hello`,
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc:     "hit the route with wrong header",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"k": "jack"},
+			Path:         `/hello`,
+			ExpectStatus: http.StatusNotFound,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc:     "hit the route with no header",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         `/hello`,
+			ExpectStatus: http.StatusNotFound,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc: "update route with vars (cookie)",
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+					"uri": "/hello",
+					"vars": [
+						["http_cookie","==","_octo=GH1.1.572248189.1598928545; _device_id=2c1a1a52074e66a3a008e4b73c690500; logged_in=yes;"]
+					],
+					"upstream": {
+						"type": "roundrobin",
+						"nodes": [{
+							"host": "172.16.238.20",
+							"port": 1980,
+							"weight": 1
+						}]
+					}
+				}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+
+		{
+			caseDesc:     "hit the route with right Cookie",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"Cookie": "_octo=GH1.1.572248189.1598928545; _device_id=2c1a1a52074e66a3a008e4b73c690500; logged_in=yes;"},
+			Path:         `/hello`,
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc:     "hit the route with wrong Cookie",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"Cookie": "jack"},
+			Path:         `/hello`,
+			ExpectStatus: http.StatusNotFound,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc:     "hit the route with no Cookie",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         `/hello`,
+			ExpectStatus: http.StatusNotFound,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc:     "delete route",
+			Object:       MangerApiExpect(t),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/routes/r1",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc: "create route with multiple vars (args, cookie and header)",
```

Comment:
update

Reply:
because the old route was deleted at previous test case (to avoid affecting this case), so I think here should be `add` again